web/src/main/java/com/marklogic/hub/web/service/FlowManagerService.java

```diff
@@ -221,14 +224,26 @@ public void deleteStep(String flowName, String stepId) {
 
     }
 
-    public String runFlow(String flowName, String[] steps) {
+    public Flow runFlow(String flowName, List<String> steps) {
         RunFlowResponse resp = null;
-        if (steps == null) {
+        if (steps == null || steps.size() ==0 ) {
             resp = flowRunner.runFlow(flowName);
         } else {
-            resp = flowRunner.runFlow(flowName, Arrays.asList(steps));
+            List<String> restrictedSteps = new ArrayList<>();
+            steps.forEach((step) -> restrictedSteps.add(this.getStepKeyInStepMap(flowName, step)));
+            resp = flowRunner.runFlow(flowName, restrictedSteps);
         }
-        return resp.getJobId();
+        return this.getFlow(flowName);
+    }
+
+    public Flow stop(String flowName) {
+        if (flowRunner.getRunningFlow().equals(flowName)) {
+            flowRunner.stopJob(flowRunner.getRunningJobId());
+        }
+        else{
```

Comment:
We also need to handle flows that are not running but are still part of the job queue.

Reply:
I spoke with @alexsmr,  he mentioned that from UI a flow cannot be started again until previous run completes, so I don't think we have to handle that case.

Reply:
Okay got it.